### ISSUE TITLE
P(r) crashing on Sasview on use of backspace key (Trac #633)

### ISSUE THREAD

**Issue by ajj** [SasView/sasview#765]
From Paul's testing on 28th August 2016:

"PR mostly works as it did except that I was able to crash !SasView (just exits abruptly and completely - no error message either popping up or in logs), by using the backspace key in the Dmax box ....sometimes ... I think I can reproduce with a few tries but have not nailed exactly the magic combination that does that."

There is a long history of crashes with P(r). Needs some debugging.

Migrated from http://trac.sasview.org/ticket/633
```json
{
    "status": "closed",
    "changetime": "2016-08-31T15:10:33",
    "_ts": "2016-08-31 15:10:33.695586+00:00",
    "description": "From Paul's testing on 28th August 2016:\n\n\"PR mostly works as it did except that I was able to crash !SasView (just exits abruptly and completely - no error message either popping up or in logs), by using the backspace key in the Dmax box ....sometimes ... I think I can reproduce with a few tries but have not nailed exactly the magic combination that does that.\"\n\nThere is a long history of crashes with P(r). Needs some debugging.",
    "reporter": "ajj",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasView Bug Fixing",
    "time": "2016-08-29T19:51:03",
    "component": "SasView",
    "summary": "P(r) crashing on Sasview on use of backspace key",
    "priority": "major",
    "keywords": "",
    "milestone": "SasView 4.0.0",
    "owner": "",
    "type": "defect"
}
```

**Comment by butlerpd** [comment]
Trac update at `2016/08/29 22:11:59`: 
* **butler** commented:

Actually we (myself but mostly Mathieu) spent some time working on the bugginess of P(r) for one of the 3.x releases (can't remember which) and fixed quite a number of things and as far as I know it has been fairly stable since.  this is a very particular key stroke combination I think -- otherwise worked fairly well.

However this is one of if not the oldest part of the code that has been hardly touched in 10 years so could certainly do with some TLC overall.


* **butler** changed description from:

> From Paul's testing on 28th August 2016:
> 
> "PR mostly works as it did except that I was able to crash SasView just exit abruptly and completely, by using the backspace key in the Dmax box ....sometimes ... think I can reproduce with a few tries but have not nailed exactly the magic combination that does that."
> 
> There is a long history of crashes with P(r). Needs some debugging.

to:

> From Paul's testing on 28th August 2016:
> 
> "PR mostly works as it did except that I was able to crash !SasView (just exits abruptly and completely - no error message either popping up or in logs), by using the backspace key in the Dmax box ....sometimes ... I think I can reproduce with a few tries but have not nailed exactly the magic combination that does that."
> 
> There is a long history of crashes with P(r). Needs some debugging.

**Comment by butlerpd** [comment]
Trac update at `2016/08/31 01:10:14`: **butler** commented:

I have traced the error down. Fixing is another story but shouldn't be too hard.  Basically P(r) is trying to be clever so is constantly recalculating stuff in the background for every change in the entry fields.  If D max becomes 0 as opposed to "empty" then there is a SVD error thrown.  Interestingly this causes the distributed version to exit instantly and without any errors logged while doing so from the developer version does nothing visible to the user at all and just keeps working ... but does register the error in the console (but maybe not in the log file).

This is thoroughly reproducible and now makes sense. So just need to capture when the entry is 0 and not allow calculations to proceed.

**Comment by ajj** [comment]
Trac update at `2016/08/31 15:10:33`: 
* **ajj** commented:

Fixed in 4036cb0


* **ajj** changed resolution from "" to "fixed"
* **ajj** changed status from "new" to "closed"
